src/Compilers/Core/MSBuildTaskTests/TargetTests.cs

```diff
@@ -0,0 +1,479 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+// uncomment the below define to dump binlogs of each test
+// #define DUMP_MSBUILD_BIN_LOG
+
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Xml;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Execution;
+using Microsoft.Build.Framework;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.BuildTasks.UnitTests
+{
+    public class TargetTests
+    {
+        [Fact]
+        public void GenerateAnalyzerConfigShouldNotRunWhenNoPropertiesOrMetadata()
```

Comment:
@rainersigwald do you think you could could cast your expert eye over the approach these tests take? They basically use the MSBuild apis to do an in-proc build of the test project and check the state of various collections after executing.

They stub out `WriteLinesToFile` (the only place we touch the disk in the targets we're testing here) so that they should be side effect free. 

Ideally i'd like to expand our testing of other targets to do something similar, so want to make sure this isn't a _totally_ crazy idea.